llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -1015,6 +1017,63 @@ void SpirvLowerGlobal::lowerInOutInPlace() {
   }
 }
 
+// =====================================================================================================================
+// @param builtIn : BuiltIn value
+// @param elemIdx : Element Index of struct
+Value *SpirvLowerGlobal::createRaytracingBuiltIn(BuiltIn builtIn) {
+  switch (builtIn) {
+  case BuiltInLaunchIdKHR:
+    return m_builder->create<DispatchRaysIndexOp>();
+  case BuiltInLaunchSizeKHR:
+    return m_builder->create<DispatchRaysDimensionsOp>();
+  case BuiltInWorldRayOriginKHR:
+    return m_builder->create<WorldRayOriginOp>();
+  case BuiltInWorldRayDirectionKHR:
+    return m_builder->create<WorldRayDirectionOp>();
+  case BuiltInObjectRayOriginKHR:
+    return m_builder->create<ObjectRayOriginOp>();
+  case BuiltInObjectRayDirectionKHR:
+    return m_builder->create<ObjectRayDirectionOp>();
+  case BuiltInRayTminKHR:
+    return m_builder->create<RayTminOp>();
+  case BuiltInRayTmaxKHR:
+    return m_builder->create<RayTcurrentOp>();
```

Comment:
The naming here is a bit unfortunate, but that is not on your side.

Reply:
> The naming here is a bit unfortunate, but that is not on your side.

ohm, i guess some of namings comes from dx spec.

Reply:
Yeah, the dialect is a bit of an eclectic mix of DXIL and SPIR-V influences.